beacon_node/network/src/beacon_processor/work_reprocessing_queue.rs

```diff
@@ -84,6 +88,8 @@ pub enum ReprocessQueueMessage<T: BeaconChainTypes> {
     UnknownBlockAggregate(QueuedAggregate<T::EthSpec>),
     /// A light client optimistic update that references a parent root that has not been seen as a parent.
     UnknownLightClientOptimisticUpdate(QueuedLightClientUpdate<T::EthSpec>),
+    /// A new backfill batch that needs to be scheduled for processing.
+    BackfillSync(WorkEvent<T>),
```

Comment:
this is not critical here since we will never have more than one queued, but the reason why we handle individual types for scheduled work is size. So a backfill batch work is 38 bytes if I'm not mistaken, and a full work event is 240bytes. So this is using per item more than 6 times what it needs to

Reply:
ah good idea to consider size of queue items, I think this may be important - if I'm not mistaken we are more likely to have more than one queued here in the reprocess queue (hence the need to rate limit it), and less likely to have more than one in the `BeaconProcessor` queue with this set up (releasing one batch every few seconds).

I'll update this to store the `process_id` and `blocks` only.

Reply:
I've pushed a fix for this, although this nested destructuring done in the below line looks a bit ugly, but I can't think of a nicer way to extract the values here

https://github.com/sigp/lighthouse/pull/3936/files#diff-3da26b4fa402b6f54c490b4e3ec1cff1971712f08549730e47be198b6bf04fa2R1066

Reply:
not close to ideal but less ugly would be to have a function similar to what you had before (`is_backfill`) that returns an option with the inner values

Reply:
also, we will _never_have more than one batch queued, because sync waits for the result of the current batch to decide if it sends a next one. This is why I considered of lesser importance